feat(noticias-endpoint): agregado un endpoint de noticias

Im-Fran opened the pull request:
Para una mejor implementación de la sección de noticias en la app he agregado este endpoint de noticias. Simplemente, va a la API del wordpress del sitio web utem.cl y obtiene:
- ID
- Fecha (en formato local con la zona America/Santiago)
- Título
- Subtitulo (que en realidad se toma como un 'resumen' en la api de wordpress)
- Link
- Imagen

Im-Fran committed:
feat(noticias-endpoint): agregado un endpoint de noticias

Para una mejor implementación de la sección de noticias en la app he agregado este endpoint de noticias. Simplemente, va a la API del wordpress del sitio web utem.cl y obtiene:
- ID
- Fecha (en formato local con la zona America/Santiago)
- Título
- Subtitulo (que en realidad se toma como un 'resumen' en la api de wordpress)
- Link
- Imagen

mapacheverdugo commented on src/utem/services/noticias.service.ts:
```diff
@@ -0,0 +1,35 @@
+export class NoticiasService {
+  static async getNoticias(porPagina: number){
+    if(porPagina > 20) {
+      porPagina = 20;
```
A que se debe esta condición?

Im-Fran replied:
Mas que nada el tiempo de carga, 20 noticias ya son 2 segundos, además que obtiene todos los datos de cada post, por lo que podría limitar recursos (realmente no lo he probado pero creo que es una posibilidad). Además ya 20 noticias son de marzo de este año, realmente no veo la necesidad de más de 20 pero podemos eliminarla, o limitar a 100. 100 noticias ya son noticias de 2021. O quizá ver si se puede obtener por rango de fechas.

mapacheverdugo commented on src/utem/services/noticias.service.ts:
```diff
@@ -0,0 +1,35 @@
+
+      const noticia: Noticia = {
+        id: post.id,
+        fecha: post.date,
```
Creo que a nivel de backend siempre es mejor ocupar la hora GMT/UTC y que a nivel de frontend se haga la transformación a la zona horaria del mismo dispositivo, por lo que usaría `post.date_gmt` y le agregaría una `Z` al final para que el string tenga formato ISO correcto

Algo como: `post.date_gmt + 'Z'`

Im-Fran replied:
me gusta esa idea, la aplicaré cuando vaya a editar esta PR

mapacheverdugo commented on src/utem/services/noticias.service.ts:
```diff
@@ -0,0 +1,35 @@
+import Noticia from "../models/noticia.model";
+import * as cheerio from 'cheerio'
+
+export class NoticiasService {
```
El servicio debería contener el nombre del dominio, como lo pusiste en `/utem`, debería llamarse `UtemNoticiasServices`

Im-Fran replied:
okok, lo cambio para la próxima edición ;)

Im-Fran committed:
patch(noticias-endpoint): arreglado comentarios de @mapacheverdugo

* Renombrado de NoticiasService a UtemNoticiasService
* Removido limite de pagina
* Agregado rango de fecha. Por defecto 6 meses atrás y hasta hoy
* Ahora los datos retornan fecha gmt en formato ISO8601

(Además agregué dayjs v1.11.9 y unas configuraciones iguales a las de la PR #9 para evitar conflictos futuros)

Im-Fran committed:
patch(noticias-endpoint): arreglado comentarios de @mapacheverdugo

* Renombrado de NoticiasService a UtemNoticiasService
* Removido limite de pagina
* Agregado rango de fecha. Por defecto 6 meses atrás y hasta hoy
* Ahora los datos retornan fecha gmt en formato ISO8601

(Además agregué dayjs v1.11.9 y unas configuraciones iguales a las de la PR #9 para evitar conflictos futuros)

Im-Fran committed:
Merge remote-tracking branch 'origin/feat/noticias-endpoint' into feat/noticias-endpoint

Im-Fran committed:
Merge branch 'dev' into 'feat/noticias-endpoint'

Im-Fran committed:
Merge branch 'dev' into feat/noticias-endpoint

# Conflicts:
#	.pnp.cjs
#	src/core/routes/index.routes.ts
#	yarn.lock

Im-Fran committed:
patch(yarn v4): se ha actualizado a yarn v4

Im-Fran committed:
patch(docker-compose): mejorado docker compose para usar yarn start-dev en lugar de yarn start

Im-Fran committed:
patch(github workflows & tests): agregado valores por defecto y variables faltantes.

Im-Fran committed:
patch: mejora de docker compose
